SyncTests/TestBookmarkModel.swift

```diff
@@ -0,0 +1,183 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Deferred
+import Foundation
+import Shared
+@testable import Storage
+@testable import Sync
+import XCTest
+
+// Thieved mercilessly from TestSQLiteBookmarks.
+private func getBrowserDBForFile(filename: String, files: FileAccessor) -> BrowserDB? {
+    let db = BrowserDB(filename: filename, files: files)
+
+    // BrowserTable exists only to perform create/update etc. operations -- it's not
+    // a queryable thing that needs to stick around.
+    if !db.createOrUpdate(BrowserTable()) {
+        return nil
+    }
+    return db
+}
+
+class TestBookmarkModel: FailFastTestCase {
+    let files = MockFiles()
+
+    override func tearDown() {
+        do {
+            try self.files.removeFilesInDirectory()
+        } catch {
+        }
+        super.tearDown()
+    }
+
+    private func getBrowserDB(name: String) -> BrowserDB? {
+        let file = "TBookmarkModel\(name).db"
+        print("DB file named: \(file)")
+        return getBrowserDBForFile(file, files: self.files)
+    }
+
+    func getSyncableBookmarks(name: String) -> MergedSQLiteBookmarks? {
+        guard let db = self.getBrowserDB(name) else {
+            XCTFail("Couldn't get prepared DB.")
+            return nil
+        }
+
+        return MergedSQLiteBookmarks(db: db)
+    }
+
+    func testBookmarkEditableIfNeverSyncedAndEmptyBuffer() {
+        guard let bookmarks = self.getSyncableBookmarks("A") else {
+            XCTFail("Couldn't get bookmarks.")
+            return
+        }
+
+        // Set a local bookmark
+        let bookmarkURL = "http://AAA.com".asURL!
+        bookmarks.local.insertBookmark(bookmarkURL, title: "AAA", favicon: nil, intoFolder: BookmarkRoots.MenuFolderGUID, withTitle: "").succeeded()
+
+        XCTAssertTrue(bookmarks.isMirrorEmpty())
```

Comment:
You're missing a `.value()` here to resolve the deferred. Look for this error throughout.